library/src/main/java/com/macroyau/thingspeakandroid/ThingSpeakChannel.java

```diff
@@ -113,16 +113,38 @@
      * @param channelId The ID of this specific Channel.
      */
     public ThingSpeakChannel(long channelId) {
-        this(channelId, null);
+        this(channelId, null, null);
     }
 
     /***
-     * Constructor for private Channels.
+     * Constructor for public Channels.
+     *
+     * @param channelId The ID of this specific Channel.
+     * @param API The address of the custom thingspeak server API.
+     */
+    public ThingSpeakChannel(long channelId, URL API) {
+        this(channelId, null, API);
+    }
+
+    /***
+     * Constructor for public Channels.
      *
      * @param channelId The ID of this specific Channel.
      * @param readApiKey The Read API Key for this specific Channel.
      */
     public ThingSpeakChannel(long channelId, String readApiKey) {
+        this(channelId, readApiKey, null);
+    }
+
+    /***
+     * Constructor for private Channels.
+     *
+     * @param channelId The ID of this specific Channel.
+     * @param readApiKey The Read API Key for this specific Channel.
+     * @param API The address of the custom thingspeak server API.
+     */
+    public ThingSpeakChannel(long channelId, String readApiKey, @Nullable URL API) {
+        THINGSPEAK_API = API.toString();
```

Comment:
You can simply check if the `api` parameter is `null`. If yes, set the Retrofit endpoint to the default one, otherwise use the custom parameter.